### PR TITLE
Instructions for OIDC login on frontend

### PR DESCRIPTION
**Pull request by eliperelman**

### CODE REVIEW

File: src/manual/using/integration/frontend.md [hunk 1]
```diff
@@ -32,9 +32,292 @@ Taskcluster API and refresh when necessary.
 
 The
 [taskcluster-oidc-test](https://github.com/taskcluster/taskcluster-oidc-test)
-repository provides an example of a sipmle SPA following this process.  You can
+repository provides an example of a simple SPA following this process.  You can
```
Comment (djmitche) [inline src/manual/using/integration/frontend.md]: It looks like that repo is a bit behind -- not using tc-client-web, etc.  It's not really even simple.. should we still point to it?

File: src/manual/using/integration/frontend.md [hunk 2]
```diff
@@ -32,9 +32,292 @@ Taskcluster API and refresh when necessary.
...
+
+Authorizing your application to use Taskcluster APIs from a frontend project is relatively
+straightforward, but there can be some hurdles to jump through. First, let's talk about a
+few prerequisites:
+
+* Possess an [auth0 Client](https://auth0.com/docs/clients) which can access Taskcluster
+credentials. If you don't have one, you can request a [New Single Sign On Application in
+The Hub](https://mozilla.service-now.com/sp?id=sc_cat_item&sys_id=1e9746c20f76aa0087591d2be1050ecb).
+Since this is going to be used on the frontend, make sure to request a client that uses
+"RS256 algorithms" and "OpenIDConnect (OIDC)".
```
Comment (djmitche) [inline src/manual/using/integration/frontend.md]: Also mention this should be a single-page application, and that RS256 and single-page need to be specified in the comments at the bottom of the form.

File: src/manual/using/integration/frontend.md [hunk 3]
```diff
@@ -32,9 +32,292 @@ Taskcluster API and refresh when necessary.
...
+<script>
+'use strict';
+
+const link = document.getElementById('action-link');
+const status = document.getElementById('status');
+</script>
+```
+
+Now we are going to create an instance of a `WebAuth` from auth0. This is
+entity that lets us communicate with the the auth0 service. This `WebAuth`
```
Comment (djmitche) [inline src/manual/using/integration/frontend.md]: *the entity

File: src/manual/using/integration/frontend.md [hunk 4]
```diff
@@ -32,9 +32,292 @@ Taskcluster API and refresh when necessary.
...
+      // If we didn't have an auth result, something went wrong.
+      // At a minimum, let's inform the user about the problem.
+      status.innerText = 'Authentication is missing payload';
+    } else {
+      // The user signed in successfully!
+      // Let's save the accessToken, update the UI,
+      // and tell the user
+      accessToken = result.accessToken;
+      status.innerText = 'Authorization successful';
+      link.innerText = 'Get Taskcluster scopes';
```
Comment (djmitche) [inline src/manual/using/integration/frontend.md]: Let's name the button "Call Taskcluster API" -- I don't want readers to think they need to call `auth.currentScopes`.

File: src/manual/using/integration/frontend.md [hunk 5]
```diff
@@ -32,9 +32,292 @@ Taskcluster API and refresh when necessary.
...
+    }
+  });
+}
+```
+
+Great, now with our user logged in, and an access token to identify them,
+we can now authenticate with Taskcluster. In the previous step we
```
Comment (djmitche) [inline src/manual/using/integration/frontend.md]: "we can now call Taskcluster API methods.  We'll call [Auth.currentScopes](https://docs.taskcluster.net/reference/platform/taskcluster-auth/references/api#currentScopes) as an example."

File: src/manual/using/integration/frontend.md (hunk 1 above)
Comment (eliperelman) [reply src/manual/using/integration/frontend.md]: I'm fine with removing it.

**Comment by eliperelman** [comment]
Changes addressed, feel free to merge when you're happy.

**Pull request merged**